### PR TITLE
Fixes #5952 Prevent adding not public CPT posts to cache table

### LINKED ISSUE #5952: WPCode URLs added to cache table when we disable active snippet
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version => 3.13.3
- Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
Whenever we disable snippet using WP Code Lite, we will add URL entry to the database (duplicated entry for 404 URL and author)

**To Reproduce**

**Precondition**
- Active theme 2020
- WPCode Lite 2.0.11 is activated
- Add some PHP snippets, enable some of them

**Steps to reproduce the behavior:**
1. Install and activate WPR 3.13.3
2. Disable one of the snippets and notice the cache table =>URL domain/wpcode/snippetName added to cache table twice and author link added twice

**Expected behavior**
- WPcode URLs not added to cache table and no duplicated entry for the author URL

**Screenshots**
Screenshot from 2023-05-26 11-36-35

**Additional context**
- tested on https://newer.rocketlabsqa.ovh/

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by Khadreal**
# Description

Fixes #5952 

## Documentation
Added a check for post type before adding it to wpr_cache_table

### User documentation

*Explain how this code impacts users.*

### Technical documentation

*Explain how this code works. Diagram & drawings are welcomed.*

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue).

## New dependencies
*List any new dependencies that are required for this change.*

## Risks
*List possible performance & security issues or risks, and explain how they have been mitigated.*

# Checklists

## Code style
- [x] I wrote self-explanatory code about what it does.
- [x] I named variables and functions explicitely.
- [x] I did not introduce unecessary complexity.

### CODE REVIEW

File: inc/common/purge.php [hunk 1]
```diff
@@ -302,9 +302,10 @@ function rocket_clean_post_cache_on_status_change( $post_id, $post_data ) {
 
 	$purge_urls = [];
 	$post       = get_post( $post_id );
+	$post_type  = get_post_type_object( $post->post_type );
 
 	// Return if $post is not an object.
-	if ( ! is_object( $post ) ) {
+	if ( ! is_object( $post ) || true !== $post_type->public ) {
```
Comment (MathieuLamiot) [inline inc/common/purge.php:308]: Either the comment needs to be adapted, either the post_type bail-out should be done in a dedicated code block ; for clarity.

**Review by MathieuLamiot — DISMISSED** [review]
We would need tests on the new bail-out.
Also, I asked on the issue if excluding private post types is expected or not. -> OK on this.

File: tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php
Comment (remyperona) [inline tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php:1]: The filename doesn't match the function name

File: tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Test class covering ::rocket_clean_post_cache_on_status_change
+ * @uses   ::rocket_clean_files
+ *
+ * @group  Common
+ * @group  Purge
+ */
+class Test_RocketCleanCachePostOnStatusChange extends TestCase {
+	protected $path_to_test_data = 'rocketCleanCacheOnStatusChange.php';
```
Comment (remyperona) [inline tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php]:
```suggestion
```

File: tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+	/**
+	 * @dataProvider providerTestData
```
Comment (remyperona) [inline tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php]:
```suggestion
	 * @dataProvider configTestData
```

File: tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+
+/**
+ * Test class covering ::rocket_clean_post_cache_on_status_change
+ * @uses   ::rocket_clean_files
+ *
+ * @group  Common
+ * @group  Purge
+ */
+class Test_RocketCleanCachePostOnStatusChange extends TestCase {
```
Comment (remyperona) [inline tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php]:
```suggestion
class TestRocketCleanPostCacheOnStatusChange extends TestCase {
```

File: tests/Unit/inc/common/rocketCleanPostCacheOnSlugChange.php [hunk 5]
```diff
@@ -97,4 +97,8 @@ public function testShouldFireRocketCleanFilesWhenExistingPostStatusIsCorrect()
 
 		$this->assertNull( rocket_clean_post_cache_on_slug_change( $post_id, [ 'post_name' => $post_name ] ) );
 	}
+
+	public function testShouldNotFireRocketCleanFilesWhenExistingPostStatusIsCorrect(){
+
+	}
```
Comment (remyperona) [inline tests/Unit/inc/common/rocketCleanPostCacheOnSlugChange.php]: What is this?

File: tests/Fixtures/inc/common/rocketCleanCacheOnStatusChange.php
Comment (remyperona) [inline tests/Fixtures/inc/common/rocketCleanCacheOnStatusChange.php:1]: The filename needs to be updated to match the correct function name

**Review by remyperona — CHANGES_REQUESTED** [review]

File: tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php (hunk 4 above)
Comment (Khadreal) [reply tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php]: This throw an error, **Class rocketCleanCachePostOnStatusChange could not be found in /var/www/html/wp-content/plugins/wp-rocket/tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php**

File: tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php
Comment (Khadreal) [reply tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php:1]: Resolved

File: tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php (hunk 4 above)
Comment (remyperona) [reply tests/Unit/inc/common/rocketCleanCachePostOnStatusChange.php]: You need to update the class name and the filename too. Notice that rocketClean**CachePost**OnStatusChange doesn't match rocket_clean_**post_cache**_on_status_change.

You'll also need to update the fixtures filename

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by MathieuLamiot** [comment]
I reproduced the issue on my E2E website with develop, and not with this PR.
I get the same number of preload entries on this PR than on develop without code snippets, no errors and all are preloaded.

**Pull request merged**